docs/conceptual/runtime/contracts/contracts_module.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Contracts Module
+---
+
+The [SRML Contracts module](https://substrate.dev/rustdocs/master/srml_contracts/index.html)
+provides the ability for the runtime to deploy and execute [WebAssembly
+(Wasm)](https://webassembly.org/) smart contracts.
+
+## Wasm Engine
+
+The SRML Contracts module depends on a Wasm sandboxing interface defining the Wasm execution engine
+available within the runtime. This is currently implemented with
+[`wasmi`](https://github.com/paritytech/wasmi), a Wasm interpreter.
+
+## Features
+
+The SRML Contracts module has a number of familiar and new features for the deployment and execution
+of smart contracts.
+
+### Account Based
+
+The Contracts module uses an account-based system similar to many existing smart contract platforms.
+To the Substrate runtime, contract accounts are just like normal user accounts; however, in addition
+to an `AccountID` and `Balance` that normal accounts have, a contract account also has associated
+contract code and some persistent contract storage.
+
+### Two Step Deployment
+
+Deploying a contract with the Contracts module takes two steps:
+
+1. Store the Wasm contract on the blockchain.
+2. Instantiate a new account, with new storage, associated with that Wasm contract.
+
+This means that multiple contract instances, with different constructor arguments, can be
+initialized using the same Wasm code, reducing the amount of storage space needed by the Contracts
+module on your blockchain.
+
+### Runtime Environment Types
+
+For writing contracts and interacting with the runtime, a set of types are available (e.g.
+`AccountId`, `Balance`, `Hash`, `Moment`). These types can be user defined for custom runtimes, or
+the supplied defaults can be used.
+
+### Contract Calls
+
+Calls to contracts can alter the storage of the contract, create new contracts, and call other
+contracts. Because Substrate provides you with the ability to write custom runtime modules, the
+Contracts module also enables you to make asynchronous calls directly to those runtime functions on
```

Comment:
async in this context means _not_ synchronous, i.e. fire and forget.

am I using the word wrong here?

If I am not mistaken, the feature exists. Whether or not you want to design developer docs for it is one thing, but if it is something we support then we should mention it in our conceptual documentation.

Reply:
@Robbepop , For now they are async / fire and forget, with the idea of making them sync as soon as we get all required infrastructure in place. I am ok leaving it as it since it is the status quo.